database/interface_cache.go

```diff
@@ -159,22 +159,22 @@ func (i *Interface) checkCache(key string) record.Record {
 	return nil
 }
 
-func (i *Interface) updateCache(r record.Record, write bool) (written bool) {
+// updateCache updates an entry in the
+func (i *Interface) updateCache(r record.Record, write bool, remove bool, ttl int64) (written bool) {
```

Comment:
:thinking: there's something missing in the comment I guess. Also, I'd document locking behavior for the record here